tests/unit/db/result_tests.py

```diff
@@ -88,43 +106,522 @@ def test_constructor(self):
+                     'id': 'julia{0:03d}'.format(x),
+                     'value': 1} for x in range(100)]
+        self.assertEqual(result[:102], expected)
+
+        result = Result(self.view001, limit=20)
+        expected = [{'key': 'julia000', 'id': 'julia000', 'value': 1},
+                    {'key': 'julia001', 'id': 'julia001', 'value': 1}]
+        self.assertEqual(result[:2], expected)
+        expected = [{'key': 'julia{0:03d}'.format(x), 
+                     'id': 'julia{0:03d}'.format(x),
+                     'value': 1} for x in range(20)]
+        self.assertEqual(result[:22], expected)
+
+        result = Result(self.view001, skip=98)
+        expected = [{'key': 'julia098', 'id': 'julia098', 'value': 1},
+                    {'key': 'julia099', 'id': 'julia099', 'value': 1}]
+        self.assertEqual(result[:2], expected)
+        self.assertEqual(result[:4], expected)
+
+        result = Result(self.view001, limit=2, skip=20)
+        expected = [{'key': 'julia020', 'id': 'julia020', 'value': 1},
+                    {'key': 'julia021', 'id': 'julia021', 'value': 1}]
+        self.assertEqual(result[:2], expected)
+        self.assertEqual(result[:4], expected)
+
+    def test_get_item_key_slice_using_invalid_options(self):
+        """
+        Test that when "key" and/or "keys" are used in the result an exception
+        is raised.
+        """
+        result = Result(self.view001, key='foo')
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo':]
+        self.assertEqual(cm.exception.status_code, 102)
+
+        result = Result(self.view001, keys=['foo', 'bar'])
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo':]
+        self.assertEqual(cm.exception.status_code, 102)
+
+        result = Result(self.view001, startkey='foo')
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo':]
+        self.assertEqual(cm.exception.status_code, 102)
+
+        result = Result(self.view001, endkey='foo')
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo':]
+        self.assertEqual(cm.exception.status_code, 102)
+
+    def test_get_item_invalid_key_slice(self):
+        """
+        Test that when invalid start and stop values are provided in a slice
+        an exception is raised.  Specifically this happens when the slice start
+        and stop are different types.
+        """
+        result = Result(self.view001)
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo': ['bar', 'baz']]
+        self.assertEqual(cm.exception.status_code, 101)
+
+        ten = ResultByKey(10)
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo': ten]
+        self.assertEqual(cm.exception.status_code, 101)
+
+    def test_get_item_key_slice_using_start_stop(self):
```

Comment:
Since this logic works as expected, I added tests to confirm the functionality.  Done in e6d1b8b.